index.js

```diff
@@ -15,8 +15,15 @@ function buildInspectorUrl(inspectorHost, inspectorPort, debugPort, fileToShow)
     hostname: inspectorHost || '127.0.0.1',
     port: inspectorPort,
     pathname: '/debug',
-    search: '?port=' + debugPort
+    query: {
+        port: debugPort
+    }
   };
+  // For backward compatibility, we'd like to keep arguments.
+  // But if arguments.length > 4, we take fifth argument as a debugHost.
+  if (buildInspectorUrl.arguments.length > 4 && buildInspectorUrl.arguments[4]) {
```

Comment:
To be honest, I haven't encountered this pattern before. We usually use `arguments` instead of `buildInspectorUrl.arguments`.

Is there any particular reason for using `arguments` instead of adding a new argument to the function signature?

```js
function buildInspectorUrl(inspectorHost, inspectorPort, debugPort, fileToShow, debugHost) {
  // ...
  if (debugHost) {
    parts.query['host'] = buildInspectorUrl.arguments[4];
  }
}
```